### PR TITLE
Create separate packages for each config

### PR DESCRIPTION
**Pull request by rekmarks**
This PR separates each config into a separate package, as follows:

- The base config is still contained in `@metamask/eslint-config`
- All other configs have been moved into their own package, named `@metamask/eslint-config-CONFIG_NAME`
  - e.g. `@metamask/eslint-config-jest`
- Config source files have been renamed and in some cases modified for formatting, but the configs are functionally identical.
- Each package has its own readme and changelog, and a readme was added to the monorepo root.
  - Release steps were removed as these are AFAIK still in flux.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  root: true,
+  extends: [
+    '@metamask/eslint-config',
```
Comment (brad-decker) [inline .eslintrc.js:4]: giphy-downsized

File: package.json [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+    "lint": "yarn eslint . --ext ts,js",
+    "lint:fix": "yarn lint --fix"
+  },
   "workspaces": [
-    "packages/base"
+    "packages/base",
+    "packages/jest",
+    "packages/mocha",
+    "packages/nodejs",
+    "packages/typescript"
```
Comment (Gudahtt) [inline package.json:13]: Nit: This might be more concise
```suggestion
    "packages/*",
```

File: packages/jest/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "src/"
+  ],
+  "scripts": {
+    "test": "eslint ."
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/MetaMask/eslint-config.git"
+  },
+  "author": "Whymarrh Whitby <whymarrh.whitby@gmail.com>",
```
Comment (Gudahtt) [inline packages/jest/package.json:20]: Nit: we should probably omit the `author` field. We omit it in most packages

File: packages/mocha/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+The contents of this package were previosly published as part of [`@metamask/eslint-config`](https://npmjs.com/package/@metamask/eslint-config).
+For changes prior to version `6.0.0`, please refer to the changelog of that package.
+
+[Unreleased]:https://github.com/MetaMask/eslint-config/compare/v5.0.0...HEAD
+<!-- [6.0.0]:https://github.com/MetaMask/eslint-config/tree/v6.0.0 -->
```
Comment (Gudahtt) [inline packages/mocha/CHANGELOG.md:13]: What is this comment for :thinking:

File: packages/mocha/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+The contents of this package were previosly published as part of [`@metamask/eslint-config`](https://npmjs.com/package/@metamask/eslint-config).
+For changes prior to version `6.0.0`, please refer to the changelog of that package.
+
+[Unreleased]:https://github.com/MetaMask/eslint-config/compare/v5.0.0...HEAD
```
Comment (Gudahtt) [inline packages/mocha/CHANGELOG.md:12]: No way to compare just this subdirectory eh? :thinking: Oh well.

File: packages/mocha/CHANGELOG.md (hunk 4 above)
Comment (rekmarks) [reply packages/mocha/CHANGELOG.md:13]: Just a placeholder for the upcoming v6 release of all these packages. I can just remove it for now if we care 😛

File: packages/jest/package.json (hunk 3 above)
Comment (rekmarks) [reply packages/jest/package.json]: 4972ea2

File: packages/typescript/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    @typescript-eslint/eslint-plugin@^3.9.1 \
+    @typescript-eslint/parser@^3.9.1 \
+    @metamask/eslint-config-typescript@^5.0.0
+```
+
+```js
+module.exports = {
+  root: true,
+  extends: [
+    '@metamask/eslint-config-typescript',
```
Comment (Gudahtt) [inline packages/typescript/README.md:20]: We should probably include the base config in all of our examples as well (both the config and the install step).

Comment (rekmarks) [reply packages/typescript/README.md]: f9e6ed1

File: packages/typescript/README.md [hunk 7]
```diff
@@ -10,24 +10,38 @@ yarn add --dev \
...
+    // explicitly as the default sourceType for TS projects is 'module'
+    {
+      files: [
+        '.eslintrc.js',
+      ],
+      parserOptions: {
+        sourceType: 'script',
+      },
+    },
+  ],
```
Comment (rekmarks) [inline packages/typescript/README.md:45]: Do we still need this?

Comment (Gudahtt) [reply packages/typescript/README.md:45]: Probably not here. Seems out-of-scope of this README.

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**